Sources/XcodeCompilationDatabaseCore/XcodeCompilationDatabase.swift

```diff
@@ -0,0 +1,209 @@
+import Foundation
+
+public func shlex(_ input: String) -> [String] {
+    var result: [String] = []
+    
+    var accumulator = ""
+    let unitQuote = ""
+    var quote = unitQuote
+    var escape = true
+    
+    func evictAccumulator() {
+        if accumulator != "" {
+            result.append(accumulator)
+            accumulator = "";
+        }
+    }
+
+    func isspace(_ c: String) -> Bool {
+        return c == " "
+    }
+
+    let doubleQuote = "\""
+    let singleQuote = "\'"
+
+    for c in input {
+        let token = String(c)
+        if escape {
+            escape = false;
+            accumulator += token
+        } else if token == "\\" {
+            escape = true;
+        } else if (quote == unitQuote && token == singleQuote) ||
+                   (quote == unitQuote && token == unitQuote) {
+            quote = token
+        } else if (quote == singleQuote && token == singleQuote) ||
+                   (quote == doubleQuote  && token == doubleQuote) {
+            quote = unitQuote
+        } else if !isspace(token) || quote != unitQuote {
+            accumulator += String(token)
+        } else {
+            evictAccumulator()
+        }
+    }
+    evictAccumulator()
+    return result;
+}
+
+/// Check if the first non " " argument is clang
+func isCC(lexed: [String]) -> Bool {
+    for arg in lexed {
+        if arg == " " {
+            continue
+        }
+        // Naieve checking for clanginess
+        return arg.hasSuffix("clang") || arg.hasSuffix("clang++")
+    }
+    return false
+}
+
+/// Frontend invocations
+/// Currently, it uses frontend invocations to prevent duplication of Swift's
+/// driver. ( consider using `swiftc` instead ) 
+func isSwiftFrontend(lexed: [String]) -> Bool {
+    for arg in lexed {
+        if arg == " " {
+            continue
+        }
+        /// Naieve checking for swiftieness
+        return arg.hasSuffix("swift")
+    }
+    return false
+}
+
+func swiftFileInSwiftInvocation(lexed: [String]) -> String? {
+    var swiftSources: [String] = []
+    var objectFile: String = ""
+    for arg in lexed {
+        if arg.hasSuffix(".swift") {
+            swiftSources.append(arg)
+        } else if arg.hasSuffix(".o") {
+            objectFile = arg
+        }
+    }
+
+    if objectFile == "" {
```

Comment:
Add a comment: this assumes `-primary-file` is being used and the first file is reasonable.